### PR TITLE
MES-629 - Calculate Cat BE Test Result

### PR DESCRIPTION
**Pull request by TomBillingtonUK**
## Description
- Removed the temporary Pass and Fail options
- Added unit tests for existing functionality

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/providers/test-result/__tests__/test-result.spec.ts [hunk 1]
```diff
@@ -61,137 +40,216 @@ describe('TestResultCalculatorProvider', () => {
...
-      const testData: CatBUniqueTypes.TestData = {
-        ...initialTestData,
-      };
-      // ASSERT
-      testResultProvider.calculateCatBTestResult(testData).subscribe((result) => {
+      const testData: CatBEUniqueTypes.TestData = {};
+      testResultProvider.calculateTestResult(TestCategory.BE, testData).subscribe((result) => {
         expect(result).toBe(ActivityCodes.PASS);
         done();
       });
```
Comment (rossfauldsbjss) [inline src/providers/test-result/__tests__/test-result.spec.ts:255]: think there should be a test for anything over 15 driving faults with no serious and no dangerous.

File: src/providers/test-result/test-result.ts [hunk 2]
```diff
@@ -11,13 +11,9 @@ import { CatBEUniqueTypes } from '@dvsa/mes-test-schema/categories/BE';
...
 
-  constructor(
-    private faultCountProvider: FaultCountProvider,
-  ) {}
-
-  calculateTestResult = (category: string,
-                         testData: CatBUniqueTypes.TestData| CatBEUniqueTypes.TestData): Observable<ActivityCode> => {
+  constructor(private faultCountProvider: FaultCountProvider) {}
 
+  public calculateTestResult(category: string, testData: object): Observable<ActivityCode> {
```
Comment (rossfauldsbjss) [inline src/providers/test-result/test-result.ts:16]: This does nothing if a category other than B or B+E is passed.  In some other cases i've seen it default to processing as cat b...  not sure if it should return a fail for anything it doesn't recognise??

**Review by rossfauldsbjss — APPROVED** [review]
looks good

File: src/providers/test-result/test-result.ts (hunk 2 above)
Comment (ammarhaiderbjss) [reply src/providers/test-result/test-result.ts:16]: probably should `throw` something

**Review by ammarhaiderbjss — APPROVED** [review]
Nice

File: src/providers/test-result/__tests__/test-result.spec.ts (hunk 1 above)
Comment (TomBillingtonUK) [reply src/providers/test-result/__tests__/test-result.spec.ts:255]: There is a test for that `should return a Fail when there are 16 driving faults'`

File: src/providers/test-result/test-result.ts (hunk 2 above)
Comment (TomBillingtonUK) [reply src/providers/test-result/test-result.ts:16]: done

**Pull request merged**